src/client/context/AppServiceContext.tsx

```diff
@@ -0,0 +1,13 @@
+import { FC, createContext } from 'react';
+
+import { ServiceContainer } from '@types';
+
+export const AppServiceContext = createContext<ServiceContainer | undefined>(undefined);
+
+interface AppServiceContextProviderProps {
+  services: ServiceContainer;
+}
+
+export const AppServiceContextProvider: FC<AppServiceContextProviderProps> = ({ children, services }) => {
+  return <AppServiceContext.Provider value={services}>{children}</AppServiceContext.Provider>;
+};
```

Comment:
To maintain the currents system architecture, we should avoid UI calling services directly, we use redux actions as an adapter layer. As an alternative, we can dispatch a thunk and we can unwrap the value, with no reducer needed.

Reply:
Since we are out of time for this feature, we can leave it for now and aim to refactor it immediately after release.

Reply:
what's the logic behind not allowing hooks or ui not call services directly? It seems that just adding a redux action as an intermediary is overhead just to keep a pattern that might not make sense in all cases.

It seems like we would just use redux for the thunk action but actually not do anything with the result from it

Reply:
We are applying the [Clean Architecture](https://blog.cleancoder.com/uncle-bob/2012/08/13/the-clean-architecture.html) principles. We are trying to maintain the business logic independent of the UI. Maintaining these separation between layers, each having their own responsibilities, make code more maintainable, flexible and scalable. In this case, actions are part of our Interface Adapters layer.

If you take a look at how it was implemented before this change, it was following this pattern. Although maintaining this architecture can be seen as having too much boilerplate, we designed it this way so that we get all the benefits from it, including utilities from the tooling itself. We add validations there, since we dont want to only have them on UI side, also makes them callable by other thunks and not just directly from UI components. Right now we are missing some critical stuff that will also benefit from this structure, like error handling and analytics through middleware, and of course testing for the core logic (actions, reducers, services).

Reply:
solved on ffefab04728c259b147ce88165c34d609bd1c7b4